File: src/main/java/com/thebluehats/server/core/modules/EventVerifiersModule.java
package com.thebluehats.server.core.modules;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.google.inject.Singleton;
import com.thebluehats.server.game.managers.combat.templates.ArrowHitPlayerVerifier;
import com.thebluehats.server.game.managers.combat.templates.PlayerHitPlayerVerifier;
import com.thebluehats.server.game.managers.enchants.CustomEnchantUtils;

public class EventVerifiersModule extends AbstractModule {
    @Provides
    @Singleton
    static PlayerHitPlayerVerifier providePlayerHitPlayerVerifier(CustomEnchantUtils customEnchantUtils) {
        return new PlayerHitPlayerVerifier();
    }

    @Provides
    @Singleton
    static ArrowHitPlayerVerifier provideArrowHitPlayerVerifier(CustomEnchantUtils customEnchantUtils) {
        return new ArrowHitPlayerVerifier();
    }

    @Override
    protected void configure() {
    }
}

File: src/main/kotlin/com/thebluehats/server/core/modules/EventVerifiersModule.kt
package com.thebluehats.server.core.modules

import com.google.inject.AbstractModule
import com.google.inject.Provides
import com.google.inject.Singleton
import com.thebluehats.server.game.managers.combat.templates.ArrowHitPlayerVerifier
import com.thebluehats.server.game.managers.combat.templates.PlayerHitPlayerVerifier
import com.thebluehats.server.game.managers.enchants.CustomEnchantUtils

class EventVerifiersModule : AbstractModule() {
    override fun configure() {}

    companion object {
        @Provides
        @Singleton
        @JvmStatic
        fun providePlayerHitPlayerVerifier(customEnchantUtils: CustomEnchantUtils): PlayerHitPlayerVerifier {
            return PlayerHitPlayerVerifier()
        }

        @Provides
        @Singleton
        @JvmStatic
        fun provideArrowHitPlayerVerifier(customEnchantUtils: CustomEnchantUtils): ArrowHitPlayerVerifier {
            return ArrowHitPlayerVerifier()
        }
    }
}